engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java

```diff
@@ -409,9 +428,9 @@ Validator createValidator(ConstraintValidatorFactory constraintValidatorFactory,
 		return metaDataProviders;
 	}
 
-	private boolean checkPropertiesForBoolean(Map<String, String> properties, String propertyKey, boolean programmaticValue) {
+	private boolean checkPropertiesForFailFast(Map<String, String> properties, boolean programmaticValue) {
```

Comment:
@gunnarmorling 

So this method was in fact very specific to the failFast case. See especially the log message of the error thrown.

I'm not sure what we really want to do here:
1/ report the error only for failFast
2/ report the error for the other properties too but in this case, we should generalize the log message. And we will probably have to change the way these booleans are initialized as the current logic only works if the default is true. In the case of the traversable resolver cache, the default is true so we can't use the same logic (we would probably need to have `Boolean` with null values in case the value is not specified).

I went with 1/ for now.